cmd/nodecmd/create_aws.go

```diff
@@ -105,158 +106,214 @@ func promptKeyPairName(ec2Svc *ec2.EC2) (string, string, error) {
 			if err != nil {
-				return nil, "", "", err
+				return nil, nil, nil, err
+			} else {
+				regions = utils.SplitComaSeparatedString(userRegionList)
 			}
 		}
 	}
-	sess, err := getAWSCloudCredentials(awsProfile, region)
-	if err != nil {
-		return nil, "", "", err
-	}
-	ec2Svc := ec2.New(sess)
-	ami, err := awsAPI.GetUbuntuAMIID(ec2Svc)
-	if err != nil {
-		if strings.Contains(err.Error(), "RequestExpired: Request has expired") {
-			printExpiredCredentialsOutput(awsProfile)
+	ec2SvcMap := map[string]*ec2.EC2{}
+	amiMap := map[string]string{}
+	for _, region := range regions {
+		sess, err := getAWSCloudCredentials(awsProfile, region)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+		ec2SvcMap[region] = ec2.New(sess)
+		amiMap[region], err = awsAPI.GetUbuntuAMIID(ec2SvcMap[region])
+		if err != nil {
+			if strings.Contains(err.Error(), "RequestExpired: Request has expired") {
+				printExpiredCredentialsOutput(awsProfile)
+			}
+			return nil, nil, nil, err
 		}
-		return nil, "", "", err
 	}
-	return ec2Svc, region, ami, nil
+	return regions, ec2SvcMap, amiMap, nil
 }
 
 // createEC2Instances creates terraform .tf file and runs terraform exec function to create ec2 instances
 func createEC2Instances(rootBody *hclwrite.Body,
-	ec2Svc *ec2.EC2,
+	ec2Svc map[string]*ec2.EC2,
 	hclFile *hclwrite.File,
-	numNodes int,
-	awsProfile,
-	region,
-	ami,
-	certName,
-	keyPairName,
-	securityGroupName string,
-) ([]string, []string, string, string, error) {
-	if err := terraformaws.SetCloudCredentials(rootBody, awsProfile, region); err != nil {
-		return nil, nil, "", "", err
+	numNodes []int,
+	awsProfile string,
+	regions []string,
+	ami map[string]string,
+	instanceConf map[string]models.InstanceConfig,
+) (map[string][]string, map[string][]string, map[string]string, map[string]string, error) {
+	if err := terraformaws.SetCloudCredentials(rootBody, awsProfile, regions); err != nil {
+		return nil, nil, nil, nil, err
 	}
-	ux.Logger.PrintToUser("Creating new EC2 instance(s) on AWS...")
-	var useExistingKeyPair bool
-	keyPairExists, err := awsAPI.CheckKeyPairExists(ec2Svc, keyPairName)
-	if err != nil {
-		return nil, nil, "", "", err
+
+	if len(numNodes) == 0 {
+		var err error
+		numNodesStr, err := app.Prompt.CaptureValidatedString("How many nodes do you want to set up on AWS?. Please use comma to separate multiple numbers in case of multiple nodes", func(input string) error {
+			integers := utils.SplitComaSeparatedUInt(input)
+			if integers == nil {
+				return fmt.Errorf("invalid input")
+			}
+			return nil
+		})
+		if err != nil {
+			return nil, nil, nil, nil, err
+		}
+		numNodes = utils.SplitComaSeparatedUInt(numNodesStr)
+	}
+	if len(numNodes) != len(regions) {
+		return nil, nil, nil, nil, fmt.Errorf("number of nodes and regions should be same")
+	}
+	for i, region := range regions {
+		if entry, ok := instanceConf[region]; ok {
+			entry.NumNodes = numNodes[i]
```

Comment:
why putting number of nodes here? seems unrelated to prefix, etc. Why not putting all together in CloudConfig?

Reply:
agree. fixed